app/templates/settings.hbs

```diff
@@ -1,227 +1,231 @@
-<div class="settings" data-test-settings-content>
-  <section class="settings-section">
-    <h2 class="small-title">
-      General
-    </h2>
-    <ul class="settings-list--columns">
-      <li>
-        <SettingsSwitch
-          @active={{this.model.settings.build_pushes}}
-          @repo={{this.repo}}
-          @description="Build pushed branches"
-          @key="build_pushes"
-        />
-        <ExternalLinkTo
-          href="https://docs.travis-ci.com/user/web-ui/#build-pushed-branches"
-          title="about branch updates"
-          class="settings-tooltip"
-        >
-          <EmberTooltip @text="Read more about branch updates" />
-          <SvgImage @name="icon-help" @class="icon-help" />
-        </ExternalLinkTo>
-      </li>
-      <li>
-        <LimitConcurrentBuilds
-          @value={{this.model.settings.maximum_number_of_builds}}
-          @enabled={{this.concurrentBuildsLimit}}
-          @repo={{this.repo}}
-        />
-      </li>
-      <li>
-        <SettingsSwitch
-          @active={{this.model.settings.build_pull_requests}}
-          @repo={{this.repo}}
-          @description="Build pushed {{vcs-vocab this.repo.vcsType 'pullRequest' plural=true lower=true}}"
-          @key="build_pull_requests"
-        />
-        <ExternalLinkTo
-          href="https://docs.travis-ci.com/user/web-ui/#build-pushed-pull-requests"
-          title="about {{vcs-vocab this.repo.vcsType 'pullRequest' lower=true}} updates"
-          class="settings-tooltip"
-        >
-          <EmberTooltip @text="Read more about {{vcs-vocab this.repo.vcsType 'pullRequest' lower=true}} updates" />
-          <SvgImage @name="icon-help" @class="icon-help" />
-        </ExternalLinkTo>
-      </li>
-    </ul>
-  </section>
-  {{#if this.showAutoCancellationSwitches}}
-    <section class="settings-section auto-cancellation">
-      <h2 class="small-title display-inline">
-        Auto Cancellation
+{{#if this.isAdmin}}
```

Comment:
To be honest, I would personally prefer to redirect users with insufficient permissions to `repo.index` instead of showing the error message. As long as we don't expose any entry points that allow to visit this page, it should be expected that it's not visitable at all. Also, redirecting would prevent fetching the data required by this route (you may notice a few `404`s in browser console currently.

What do you think?

Reply:
Awesome, thank you! :)